File: pages/site/cgi/printenv.js
let headers = [];
let body;

body = "<html>\n";
body += "<title>Server-provided Environment variables</title>";
body += "<body>\n";
body += "<table>\n";
body += "<th><td colspan=2 align=center>Environment Variables</td></th>\n";
for (let key in process.env) {
    body += `<tr><td>${key}</td><td>${process.env[key]}</td></tr>\n`;
}
body += "</table></body></html>\n";

headers.push("Content-type: text/html");
headers.push(`Content-length: ${body.length}`);

headers.forEach(header => {
    console.log(header);
});
console.log();
console.log(body);

File: pages/site/cgi/printenv.py
import os
import fileinput

print "Content-type: text/html\n\n"
print "<html>\n"
print "<title>Server-provided Environment variables</title>"
print "<body>\n"
print "<table>\n"
print "<th><td colspan=2 align=center>Environment Variables</td></th>\n"
for param in os.environ.keys():
    print "<tr><td>%s</td><td>%s</td></tr>\n" % (param,os.environ[param])
print "</table>\n"

print "<label>Body: </label>\n"
for line in fileinput.input():
    print "%s\n" % line

print "</body></html>\n"
